gcp/src/main/java/org/apache/iceberg/gcp/biglake/BigLakeClientImpl.java

```diff
@@ -0,0 +1,287 @@
+                    .setParent(getDatabase(name).toString())
+                    .setTableId(name.getTable())
+                    .setTable(table)
+                    .build()));
+  }
+
+  @Override
+  public Table getTable(TableName name) {
+    return convertException(
+        () -> {
+          try {
+            return stub.getTable(GetTableRequest.newBuilder().setName(name.toString()).build());
+          } catch (PermissionDeniedException e) {
+            throw new NoSuchTableException(
+                e, "Table %s not found or permission denied", name.toString());
+          }
+        });
+  }
+
+  @Override
+  public Table updateTableParameters(TableName name, Map<String, String> parameters, String etag) {
+    Table.Builder builder = Table.newBuilder().setName(name.toString()).setEtag(etag);
+    builder.getHiveOptionsBuilder().putAllParameters(parameters);
+    return convertException(
+        () -> {
+          try {
+            return stub.updateTable(
+                UpdateTableRequest.newBuilder()
+                    .setTable(builder)
+                    .setUpdateMask(FieldMask.newBuilder().addPaths("hive_options.parameters"))
+                    .build());
+          } catch (PermissionDeniedException e) {
+            throw new NoSuchTableException(
+                e, "Table %s not found or permission denied", name.toString());
+          }
+        });
+  }
+
+  @Override
+  public Table renameTable(TableName name, TableName newName) {
+    return convertException(
+        () -> {
+          try {
+            return stub.renameTable(
+                RenameTableRequest.newBuilder()
+                    .setName(name.toString())
+                    .setNewName(newName.toString())
+                    .build());
+          } catch (PermissionDeniedException e) {
+            throw new NoSuchTableException(
+                e, "Table %s not found or permission denied", name.toString());
+          }
+        });
+  }
+
+  @Override
+  public Table deleteTable(TableName name) {
+    return convertException(
+        () -> {
+          try {
+            return stub.deleteTable(
+                DeleteTableRequest.newBuilder().setName(name.toString()).build());
+          } catch (PermissionDeniedException e) {
+            throw new NoSuchTableException(
+                e, "Table %s not found or permission denied", name.toString());
+          }
+        });
+  }
+
+  @Override
+  public Iterable<Table> listTables(DatabaseName name) {
+    return convertException(
+        () ->
+            stub.listTables(ListTablesRequest.newBuilder().setParent(name.toString()).build())
+                .iterateAll());
+  }
+
+  // Converts BigLake API errors to Iceberg errors.
+  private <T> T convertException(Supplier<T> result) {
+    try {
+      return result.get();
+    } catch (PermissionDeniedException e) {
+      throw new NotAuthorizedException(e, "Not authorized to call BigLake API");
```

Comment:
I think this is incorrect. The person may be authorized to call the API, but not authorized for the attempted action.

Reply:
Updated it to a more general message "BigLake API permission denied". The nested error `e` contains the detailed message from the server.